.github/workflows/test.yml

```diff
@@ -55,49 +54,137 @@ jobs:
               cpanminus \
               debhelper \
               fakeroot \
               libfile-which-perl \
               libtest-exception-perl
+
+      - name: Install CPAN dependencies
+        run: |
           cpanm \
               File::Which \
               Module::Build \
               Test::Cmd \
               Test::Exception
 
+      - name: Build perl
+        run: ./configure.bash
+
       - name: Build
-        run: |
-          perl --version
-          ./configure.bash
-          make
+        run: make
 
       - name: Test build
+        run: make test
+
+      - name: Buld distribution
+        run: make dist
+
+      - name: Run distribution checks
+        run: make distcheck
+
+      - name: Run distribution test
+        run: make disttest
+
+  test-package:
+    name: Test packaging Perl ${{ matrix.perl }} on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        perl:
+          - '5.34'
+          - '5.32'
+          - '5.30'
+          - '5.28'
+        include:
+          - os: ubuntu-18.04
+            release: bionic
+          - os: ubuntu-20.04
+            release: focal
+          - os: ubuntu-latest
+            release: focal
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2.4.0
+        with:
+          fetch-depth: 0
+
+      - name: Setup perl
+        uses: shogo82148/actions-setup-perl@v1
+        with:
+          perl-version: ${{ matrix.perl }}
+
+      - name: Get release codename
+        id: distro
+        run: echo "::set-output name=codename::$(lsb_release --codename --short)"
+
+      - name: Enable fetching packages from packagecloud test repository
         run: |
-          make test
-          sudo mv /dev/stdout /dev/stdout.bak
-          make test
+          curl -Ls https://packagecloud.io/linz/test/gpgkey | sudo apt-key add -
+          echo "deb https://packagecloud.io/linz/test/ubuntu/ ${{ steps.distro.outputs.codename }} main" | sudo tee /etc/apt/sources.list.d/linz-test.list
 
-      - name: Test distribution
+      - name: Enable fetching packages from packagecloud prod repository
         run: |
-          make dist
-          make distcheck
-          make disttest
+          curl -Ls https://packagecloud.io/linz/prod/gpgkey | sudo apt-key add -
+          echo "deb https://packagecloud.io/linz/prod/ubuntu/ ${{ steps.distro.outputs.codename }} main" | sudo tee /etc/apt/sources.list.d/linz-prod.list
+
+      - name: Update package database
+        run: sudo apt-get update
+
+      - name: Install debian dependencies
+        run: |
+          sudo -E apt-get --assume-yes -qq install --no-install-recommends \
+              cpanminus \
+              debhelper \
+              fakeroot \
+              libfile-which-perl \
+              libtest-exception-perl \
+              linz-bde-copy
 
-      - name: Test packaging
+      - name: Install CPAN dependencies
```

Comment:
CPAN is the package repository; this should probably be "Install Perl dependencies".